.github/workflows/on-merge.yml

```diff
@@ -0,0 +1,62 @@
+name: On merge
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
```

Comment:
Since build is not used by any job other than release-publish in this workflow, I'd be tempted to say that you can actually merge it with release-publish.

This will also address my other comment about the `actions/download-artifact`.